near-token-factory/src/lib.rs

```diff
@@ -27,11 +28,23 @@ enum StorageKey {
     TokenMap,
 }
 
+#[derive(AccessControlRole, Deserialize, Serialize, Copy, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub enum AclRole {
+    /// Accounts with this role can replace the token binary that will be used for new tokens.
+    TokenBinaryUpdater,
+    /// Accounts with this role can replace the super admin that will be used for new tokens.
+    TokenControllerUpdater,
+}
+
+#[access_control(role_type(AclRole))]
 #[near_bindgen]
 #[derive(BorshSerialize, BorshDeserialize, PanicOnDefault)]
 pub struct Contract {
     /// Account id of the engine. It is expected to be `aurora`.
     aurora: AccountId,
+    /// Account id that will be used as super admin for all deployed tokens.
```

Comment:
```suggestion
    /// Account id that will be used as super admin for newly deployed tokens.
```

Nit but I think _all_ can be misleading since changing `token_super_admin` doesn't affect tokens that were already deployed (see doc comment of `replace_token_admin`).